framework/wazuh/core/cluster/worker.py

```diff
@@ -483,6 +483,9 @@ async def recv_agent_groups_information(self, task_id: bytes, info_type: str, lo
         data = await super().get_chunks_in_task_id(task_id, error_command)
         result = await super().update_chunks_wdb(data, info_type, logger, error_command, timeout)
         response = await self.send_request(command=command, data=json.dumps(result).encode())
+        if command == b'syn_wgc_e':
+            # Recalculate group hash before comparing with master's
+            await AsyncWazuhDBConnection().run_wdb_command(command='global recalculate-agent-group-hashes')
```

Comment:
Upon further reflection, I think it would be better to keep the recalculation within the `recv_agent_groups_entire_information` method. 

It turns out that when the entire data is received, the global hash is not included (I don't remember the reason behind this decision) (`"get_global_hash": False`):
https://github.com/wazuh/wazuh/blob/22449c835383c96f7aa5a1eb65cc1837a990dccc/framework/wazuh/core/cluster/master.py#L669-L673

Therefore, the checksums will not be compared in this case and it is not necessary to recalculate it before (not) comparing with the master's. This way, we also avoid evaluating the condition in each agent-group iteration.

Also, catch any possible WazuhExceptions.

Reply:
Carried out the same tests as mentioned here after the requested changes:

- Manually setting the group_hash to NULL

Master
```
2024/05/17 10:56:00 INFO: [Worker d19690c98c41] [Agent-groups send full] Starting.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups send full] Recalculating agent-group hash.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups send full] Obtained 1 chunks of data in 0.001s.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups send full] Sending chunks.
2024/05/17 10:56:00 INFO: [Worker d19690c98c41] [Agent-groups send full] Finished in 7200.011s. Updated 1 chunks.
```

<details><summary>Worker</summary>

```
2024/05/17 10:56:00 INFO: [Worker d19690c98c41] [Agent-groups recv] Starting.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups recv] 1/1 chunks updated in wazuh-db in 0.000s.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups recv] Obtained 1 chunks of data in 0.001s.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups recv] The checksum of master (None) and worker (aac8edbccfab5ea7eee53016af08b756a44dc399) are different.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups recv] Checksum comparison failed (5/5).
2024/05/17 10:56:00 INFO: [Worker d19690c98c41] [Agent-groups recv] Sent request to obtain all agent-groups information from the master node.
2024/05/17 10:56:00 INFO: [Worker d19690c98c41] [Agent-groups recv] Finished in 0.010s. Updated 1 chunks.
2024/05/17 10:56:00 INFO: [Worker d19690c98c41] [Agent-groups recv full] Starting.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups recv full] 1/1 chunks updated in wazuh-db in 0.000s.
2024/05/17 10:56:00 INFO: [Worker d19690c98c41] [Agent-groups recv full] Finished in 0.004s. Updated 1 chunks.
2024/05/17 10:56:00 DEBUG: [Worker d19690c98c41] [Agent-groups recv full] Recalculating agent-group hash.
2024/05/17 10:56:10 INFO: [Worker d19690c98c41] [Agent-groups recv] Starting.
2024/05/17 10:56:10 DEBUG: [Worker d19690c98c41] [Agent-groups recv] 1/1 chunks updated in wazuh-db in 0.000s.
2024/05/17 10:56:10 DEBUG: [Worker d19690c98c41] [Agent-groups recv] Obtained 1 chunks of data in 0.001s.
2024/05/17 10:56:10 DEBUG: [Worker d19690c98c41] [Agent-groups recv] The checksum of both databases match. 
2024/05/17 10:56:10 INFO: [Worker d19690c98c41] [Agent-groups recv] Finished in 0.006s. Updated 1 chunks.
```

</details>

- Newly added node

Master
```
2024/05/17 11:01:50 INFO: [Worker node01] [Agent-groups send full] Starting.
2024/05/17 11:01:50 DEBUG: [Worker node01] [Agent-groups send full] Recalculating agent-group hash.
2024/05/17 11:01:50 DEBUG: [Worker node01] [Agent-groups send full] Obtained 1 chunks of data in 0.001s.
2024/05/17 11:01:50 DEBUG: [Worker node01] [Agent-groups send full] Sending chunks.
2024/05/17 11:01:50 INFO: [Worker node01] [Agent-groups send full] Finished in 7200.017s. Updated 1 chunks.
```

<details><summary>Worker</summary>

```
2024/05/17 09:01:10 INFO: [Worker node01] [Agent-groups recv] Starting.
2024/05/17 09:01:10 DEBUG: [Worker node01] [Agent-groups recv] 1/1 chunks updated in wazuh-db in 0.001s.
2024/05/17 09:01:10 DEBUG: [Worker node01] [Agent-groups recv] Obtained 1 chunks of data in 0.002s.
2024/05/17 09:01:10 DEBUG: [Worker node01] [Agent-groups recv] The checksum of master (aac8edbccfab5ea7eee53016af08b756a44dc399) and worker (None) are different.
2024/05/17 09:01:10 DEBUG: [Worker node01] [Agent-groups recv] Checksum comparison failed (1/5).
2024/05/17 09:01:10 INFO: [Worker node01] [Agent-groups recv] Finished in 0.016s. Updated 1 chunks.
2024/05/17 09:01:20 INFO: [Worker node01] [Agent-groups recv] Starting.
2024/05/17 09:01:20 DEBUG: [Worker node01] [Agent-groups recv] 1/1 chunks updated in wazuh-db in 0.000s.
2024/05/17 09:01:20 DEBUG: [Worker node01] [Agent-groups recv] Obtained 1 chunks of data in 0.001s.
2024/05/17 09:01:20 DEBUG: [Worker node01] [Agent-groups recv] The checksum of master (aac8edbccfab5ea7eee53016af08b756a44dc399) and worker (None) are different.
2024/05/17 09:01:20 DEBUG: [Worker node01] [Agent-groups recv] Checksum comparison failed (2/5).
2024/05/17 09:01:20 INFO: [Worker node01] [Agent-groups recv] Finished in 0.007s. Updated 1 chunks.
2024/05/17 09:01:30 INFO: [Worker node01] [Agent-groups recv] Starting.
2024/05/17 09:01:30 DEBUG: [Worker node01] [Agent-groups recv] 1/1 chunks updated in wazuh-db in 0.000s.
2024/05/17 09:01:30 DEBUG: [Worker node01] [Agent-groups recv] Obtained 1 chunks of data in 0.001s.
2024/05/17 09:01:30 DEBUG: [Worker node01] [Agent-groups recv] The checksum of master (aac8edbccfab5ea7eee53016af08b756a44dc399) and worker (None) are different.
2024/05/17 09:01:30 DEBUG: [Worker node01] [Agent-groups recv] Checksum comparison failed (3/5).
2024/05/17 09:01:30 INFO: [Worker node01] [Agent-groups recv] Finished in 0.007s. Updated 1 chunks.
2024/05/17 09:01:40 INFO: [Worker node01] [Agent-groups recv] Starting.
2024/05/17 09:01:40 DEBUG: [Worker node01] [Agent-groups recv] 1/1 chunks updated in wazuh-db in 0.001s.
2024/05/17 09:01:40 DEBUG: [Worker node01] [Agent-groups recv] Obtained 1 chunks of data in 0.000s.
2024/05/17 09:01:40 DEBUG: [Worker node01] [Agent-groups recv] The checksum of master (aac8edbccfab5ea7eee53016af08b756a44dc399) and worker (None) are different.
2024/05/17 09:01:40 DEBUG: [Worker node01] [Agent-groups recv] Checksum comparison failed (4/5).
2024/05/17 09:01:40 INFO: [Worker node01] [Agent-groups recv] Finished in 0.008s. Updated 1 chunks.
2024/05/17 09:01:50 INFO: [Worker node01] [Agent-groups recv] Starting.
2024/05/17 09:01:50 DEBUG: [Worker node01] [Agent-groups recv] 1/1 chunks updated in wazuh-db in 0.000s.
2024/05/17 09:01:50 DEBUG: [Worker node01] [Agent-groups recv] Obtained 1 chunks of data in 0.001s.
2024/05/17 09:01:50 DEBUG: [Worker node01] [Agent-groups recv] The checksum of master (aac8edbccfab5ea7eee53016af08b756a44dc399) and worker (None) are different.
2024/05/17 09:01:50 DEBUG: [Worker node01] [Agent-groups recv] Checksum comparison failed (5/5).
2024/05/17 09:01:50 INFO: [Worker node01] [Agent-groups recv] Sent request to obtain all agent-groups information from the master node.
2024/05/17 09:01:50 INFO: [Worker node01] [Agent-groups recv] Finished in 0.009s. Updated 1 chunks.
2024/05/17 09:01:50 INFO: [Worker node01] [Agent-groups recv full] Starting.
2024/05/17 09:01:50 DEBUG: [Worker node01] [Agent-groups recv full] 1/1 chunks updated in wazuh-db in 0.007s.
2024/05/17 09:01:50 INFO: [Worker node01] [Agent-groups recv full] Finished in 0.011s. Updated 1 chunks.
2024/05/17 09:02:00 INFO: [Worker node01] [Agent-groups recv] Starting.
2024/05/17 09:02:00 DEBUG: [Worker node01] [Agent-groups recv] 1/1 chunks updated in wazuh-db in 0.000s.
2024/05/17 09:02:00 DEBUG: [Worker node01] [Agent-groups recv] Obtained 1 chunks of data in 0.001s.
2024/05/17 09:02:00 DEBUG: [Worker node01] [Agent-groups recv] The checksum of both databases match.
```

</details>